starlette/responses.py

```diff
@@ -34,6 +35,29 @@ def guess_type(
     return mimetypes_guess_type(url, strict)
 
 
+@runtime_checkable
+class ACloseAble(Protocol):
+    async def aclose(self) -> None:  # pragma: no cover
+        ...
+
+
+T = typing.TypeVar("T")
+
+
+class aclosing(typing.Generic[T]):
+    # This is taken from 3.10 contextlib.aclosing and modified to only
+    # call thing.aclose() if it is ACloseAble.
+    def __init__(self, thing: T):
+        self.thing = thing
+
+    async def __aenter__(self) -> T:
+        return self.thing
+
+    async def __aexit__(self, *exc_info: typing.Any) -> None:
+        if isinstance(self.thing, ACloseAble):
```

Comment:
I'm not sure I see the advantage of `ACloseAble` over just `hasattr(self.thing, "aclose")`? We haven't used protocols in Starlette yet and I'm not against it but not sure it's justified here either.

Reply:
mypy doesn't know about hasattr

Reply:
To me this is doing a little too much to appease the linter.

Reply:
I've reverted these changes